source/common/router/scoped_rds.cc

```diff
@@ -103,16 +104,28 @@ ScopedRdsConfigSubscription::ScopedRdsConfigSubscription(
       Envoy::Config::SubscriptionBase<envoy::config::route::v3::ScopedRouteConfiguration>(
           scoped_rds.scoped_rds_config_source().resource_api_version(),
           factory_context.messageValidationContext().dynamicValidationVisitor(), "name"),
-      factory_context_(factory_context), name_(name), scope_key_builder_(scope_key_builder),
+      factory_context_(factory_context), name_(name),
       scope_(factory_context.scope().createScope(stat_prefix + "scoped_rds." + name + ".")),
       stats_({ALL_SCOPED_RDS_STATS(POOL_COUNTER(*scope_), POOL_GAUGE(*scope_))}),
-      rds_config_source_(std::move(rds_config_source)), stat_prefix_(stat_prefix),
-      route_config_provider_manager_(route_config_provider_manager) {
+      scope_key_builder_(scope_key_builder), rds_config_source_(std::move(rds_config_source)),
+      stat_prefix_(stat_prefix), route_config_provider_manager_(route_config_provider_manager) {
   const auto resource_name = getResourceName();
-  subscription_ =
-      factory_context.clusterManager().subscriptionFactory().subscriptionFromConfigSource(
-          scoped_rds.scoped_rds_config_source(), Grpc::Common::typeUrl(resource_name), *scope_,
-          *this, resource_decoder_, false);
+  std::unique_ptr<xds::core::v3::ResourceLocator> srds_resources_locator;
+  if (!scoped_rds.srds_resources_locator().empty()) {
+    srds_resources_locator = std::make_unique<xds::core::v3::ResourceLocator>(
+        Envoy::Config::XdsResourceIdentifier::decodeUrl(scoped_rds.srds_resources_locator()));
+  }
+  if (srds_resources_locator == nullptr) {
```

Comment:
What's the advantage of using the `srds_resources_locator` pointer? Shouldn't this be `if (scoped_rds.srds_resources_locator().empty())` instead?